vector.go

```diff
@@ -45,3 +47,13 @@ var SupportedSimilarityMetrics = map[string]struct{}{
 	EuclideanDistance: {},
 	CosineSimilarity:  {},
 }
+
+const IndexOptimizationForRecall = "recall"
+const IndexOptimizationForLatency = "latency"
+
+const DefaultIndexOptimization = IndexOptimizationForRecall
+
+var SupportedIndexConfigOptimizations = map[string]struct{}{
+	IndexOptimizationForRecall:  {},
+	IndexOptimizationForLatency: {},
```

Comment:
Set value of this map to int, so you can use in zapx where you're persisting, rather than having to declare attributes for this in multiple repos.